packages/react-reconciler/src/ReactFiber.js

```diff
@@ -874,6 +878,17 @@ export function createFiberFromOffscreen(
   fiber.stateNode = primaryChildInstance;
   return fiber;
 }
+export function createFiberFromActivity(
+  pendingProps: OffscreenProps,
+  mode: TypeOfMode,
+  lanes: Lanes,
+  key: null | string,
+): Fiber {
+  const fiber = createFiber(ActivityComponent, pendingProps, key, mode);
+  fiber.elementType = REACT_ACTIVITY_TYPE;
```

Comment:
This is not actually right. This was wrong for `Offscreen` too.

In the case of `const LazyActivity = React.lazy(async () => ({ default: React.Activity })); <LazyActivity />` the `elementType` should be the Lazy wrapper object. Otherwise reconciliation won't work correctly.

In fact, it looks like we get this wrong for all these built-ins so wrapping them would yield the wrong reconciliation.

So it's an existing bug that we need to fix separately.

Reply:
Is this wrong for `REACT_VIEW_TRANSITION_TYPE` too?

What's the fix?

Reply:
Actually, we throw for this case already with an error that says:

> [!CAUTION]
>Element type is invalid. Received a promise that resolves to: ViewTransition. Lazy element type must resolve to a class or function.


This is broken for portal, so I fixed it and tested all the built-ins here: https://github.com/facebook/react/pull/32640